omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ScheduledReportResource.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.reportingrest.web.resource;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.reporting.report.ReportRequest;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.openmrs.module.reporting.report.definition.service.ReportDefinitionService;
+import org.openmrs.module.reporting.report.service.ReportService;
+import org.openmrs.module.reportingrest.web.controller.ReportingRestController;
+import org.openmrs.module.reportingrest.web.wrapper.ScheduledReport;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.AlreadyPaged;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+@Resource(name = RestConstants.VERSION_1 + ReportingRestController.REPORTING_REST_NAMESPACE + "/scheduledReport",
```

Comment:
Can't this class be handled by hitting the `reportRequest` endpoint with appropriate parameters to only return scheduled reports?  Or can't we make it do so?

Reply:
Basically, this is `reportDefinition left join reportRequest` - one request. 
With existing APIs, we would need to REST GET `reportDefinition` and `reportRequest` then join them on FE side.

Reply:
I see.  OK.  Couldn't we accomplish this without adding the `ScheduledReport` class, and just return a List<SimpleObject> where each simple object contains a `reportDefinition` -> `ReportDefinition` and `scheduledReports` -> `List<ReportRequest>` object?

Then, in the frontend, you'd hit this endpoint and it would give you all available report definitions, and any scheduled reports for them, or an empty list if none.